mmv1/third_party/terraform/services/kms/data_source_google_kms_crypto_key_version_latest.go

```diff
@@ -0,0 +1,218 @@
+	}
+	if err := d.Set("algorithm", latestVersion["algorithm"]); err != nil {
+		return fmt.Errorf("Error setting CryptoKeyVersion: %s", err)
+	}
+
+	url, err = tpgresource.ReplaceVars(d, config, "{{KMSBasePath}}{{crypto_key}}")
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Getting purpose of CryptoKey: %#v", url)
+	res, err = transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+		Config:    config,
+		Method:    "GET",
+		Project:   cryptoKeyId.KeyRingId.Project,
+		RawURL:    url,
+		UserAgent: userAgent,
+	})
+	if err != nil {
+		return transport_tpg.HandleDataSourceNotFoundError(err, d, fmt.Sprintf("KmsCryptoKey %q", d.Id()), url)
+	}
+
+	if res["purpose"] == "ASYMMETRIC_SIGN" || res["purpose"] == "ASYMMETRIC_DECRYPT" {
+		url, err = tpgresource.ReplaceVars(d, config, "{{KMSBasePath}}{{crypto_key}}/cryptoKeyVersions/{{version}}/publicKey")
+		if err != nil {
+			return err
+		}
+		log.Printf("[DEBUG] Getting public key of CryptoKeyVersion: %#v", url)
+
+		res, err = transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+			Config:               config,
+			Method:               "GET",
+			Project:              cryptoKeyId.KeyRingId.Project,
+			RawURL:               url,
+			UserAgent:            userAgent,
+			Timeout:              d.Timeout(schema.TimeoutRead),
+			ErrorRetryPredicates: []transport_tpg.RetryErrorPredicateFunc{transport_tpg.IsCryptoKeyVersionsPendingGeneration},
+		})
+
+		if err != nil {
+			log.Printf("Error generating public key: %s", err)
+			return err
+		}
+
+		if err := d.Set("public_key", flattenKmsCryptoKeyVersionPublicKey(res, d)); err != nil {
+			return fmt.Errorf("Error setting CryptoKeyVersion public key: %s", err)
+		}
+	}
+	d.SetId(fmt.Sprintf("//cloudkms.googleapis.com/v1/%s/cryptoKeyVersions/%d", d.Get("crypto_key"), d.Get("version")))
+
+	return nil
+}
+
+func flattenKmsCryptoKeyVersionVersionLatest(v interface{}, d *schema.ResourceData) interface{} {
+	parts := strings.Split(v.(string), "/")
+	version := parts[len(parts)-1]
+	// Handles the string fixed64 format
+	if intVal, err := tpgresource.StringToFixed64(version); err == nil {
+		return intVal
+	} // let terraform core handle it if we can't convert the string to an int.
+	return v
+}
+
+func flattenKmsCryptoKeyVersionNameLatest(v interface{}, d *schema.ResourceData) interface{} {
+	return v
+}
+
+func flattenKmsCryptoKeyVersionStateLatest(v interface{}, d *schema.ResourceData) interface{} {
+	return v
+}
+
+func flattenKmsCryptoKeyVersionProtectionLevelLatest(v interface{}, d *schema.ResourceData) interface{} {
+	return v
+}
+
+func flattenKmsCryptoKeyVersionAlgorithmLatest(v interface{}, d *schema.ResourceData) interface{} {
+	return v
+}
+
+func flattenKmsCryptoKeyVersionLatest(versionsList []interface{}, d *schema.ResourceData, config *transport_tpg.Config, cryptoKeyId string) (interface{}, error) {
+	latestVersion := versionsList[len(versionsList)-1].(map[string]interface{})
```

Comment:
Does this always work? If the last key has been disabled or deleted does it show up here, or what would be considered "latest"?

Reply:
And can you add a test for such a case?

Reply:
When marking versions as `Disabled` they are still present in the list but are not available through the API, the following error appears which is also the result of the failing vcr test (will be destroyed is treated the same as a `disabled` state):

`tf output`:
```hcl
 Error: googleapi: Error 400: projects/hc-terraform-testing/locations/us-central1/keyRings/mau-latest-test/cryptoKeys/mau-crypto-key-test/cryptoKeyVersions/4 is not enabled, current state is: DESTROY_SCHEDULED.
```

The version list in gcloud console:
![image](https://github.com/user-attachments/assets/e541ae7e-5605-4fc7-bcdd-8ba9c00a1767)

So although it's marked as `disabled` it will still show up as the latest version. I'll look into adding a filter in order to get versions that have the state as `enabled`.

Reply:
Upon working on this more, I'm starting to believe that having a data source where it's only purpose is to grab the latest version seems like not the best option especially when we consider that their is currently no existing `cryptoKeyVersions.list` data source.

This realization comes from the fact that I would also need to implement `pageTokens` to cover the case of a cryptoKey containing many versions where it requires pages.

It would be best to actually just focus on a plural `cryptoKeyVersions` data source where a boolean field can be added that grabs you the latest version.

example:
```hcl
data "google_kms_crypto_key_versions" "plural" {
    crypto_key = "mau-crypto-latest-test"
    latest_version = true
}
```

@slevenick let me know your thoughts on this.

Reply:
Hmmm, google_kms_crypto_key_versions implies that it returns a collection of versions, and latest_version making that only return 1 seems unusual. Maybe the google_kms_crypto_key_versions could have a `latest_version` field that contains the latest version though.

Reply:
after a conversation with @SarahFrench through slack, we agreed that the best approach would be to first add support for a plural `crypto_key_versions` data source.

From their support for a singular `crypo_key_version_latest` data source can be then worked on where it uses the same logic from the plural data source for getting the latest version from a paginated list.

I'll be closing this and opening a PR for the plural one first. Thanks for your initial feedback on this! @slevenick